src/utils/fsOperations.ts

```
@@ -546,6 +552,100 @@ export const NodeFsOperations: FsOperations = {
+        // A close failure after writeFile completed cannot safely be interpreted
+        // as a failed append; replaying would duplicate the committed batch.
+        if (!committed && operationError === undefined) operationError = error
+      }
+      if (operationError !== undefined) throw operationError
+    } finally {
+      try {
+        await directoryHandle.close()
+      } catch (error) {
+        // Once writeFile commits, a directory-close failure cannot make replay
+        // safe; report success rather than duplicating the append batch.
+        if (!committed) throw error
```

Biome (2.5.6)

[error] Unsafe usage of 'throw'.

(lint/correctness/noUnsafeFinally)